### ISSUE TITLE
[Bug] Pressing delete key in preference triggers the deletion of the table

### ISSUE THREAD

**Issue by leprinco** [syd711/vpin-studio#566]
With a table selected, then opening preference to change my vpin name :
![image](https://github.com/user-attachments/assets/32851938-5241-4eb7-8713-fc97c145093d)

by pressing the delete key to suppress a letter, it triggers instead the deletion of the table :
![image](https://github.com/user-attachments/assets/d11a333e-cd81-4ece-a287-d095a389c3de)

**Comment by leprinco** [comment]
Fixed in 3.9
I changed an important behavior : I now install StudioKeyEventHandler as EventHandler instead of EventFilter : so instead of intercepting events from top to bottom (in the capturing phase), they are are now intercepted from bottom to top in the bubbling phase, which is to me the more natural way

So now textfields like in the ini tab or preferences dialog get the key events (especially Del key strokes) before the global studio event handler, and consume them.

I also added a KeyEventHandler at Preference Panel root that consumes any KeyEvent non intercepted by preference panels, making the global preference pane behaves as a modal window. It prevent delete, Ctrl+O or other events to bubble up and then be caught by StudioKeyEventHandler that propagate them to the context menu. While there, I profit to bind the ESC key to close the preference window.
